api/v1alpha1/vaultdynamicsecret_types.go

```diff
@@ -18,10 +18,25 @@ type VaultDynamicSecretSpec struct {
 	Namespace string `json:"namespace,omitempty"`
 	// Mount path of the secret's engine in Vault.
 	Mount string `json:"mount"`
+	// RequestHTTPMethod to use when syncing Secrets from Vault.
+	// Setting a value here is not typically required.
+	// If left unset the Operator will make requests using the GET method.
+	// In the case where Params are specified the Operator will use the PUT method.
+	// Please consult https://developer.hashicorp.com/vault/docs/secrets if you are
+	// uncertain about what method to use.
+	// +kubebuilder:validation:Enum={GET,PUT}
+	RequestHTTPMethod string `json:"requestHTTPMethod,omitempty"`
+	// uncertain about what 'path' should be set to.
 	// Path in Vault to get the credentials for, and is relative to Mount.
 	// Please consult https://developer.hashicorp.com/vault/docs/secrets if one is
 	// uncertain about what 'path' should be set to.
 	Path string `json:"path"`
+	// Params that can be passed when requesting credentials/secrets.
+	// When Params is set the configured RequestHTTPMethod will be
+	// ignored. See RequestHTTPMethod for more details.
```

Comment:
This is counter intuitive to me. `RequestHTTPMethod` is a more detailed and explicit setting, so it should take precedence. Even if they set up something weird like a GET with a body, it's not very restful but IIRC I think Vault still accepts that and treats the body parameters as though they're query parameters.

Reply:
Ah I see where this came from, leave it as it is then IMO